ssz/sedes/serializable.py

```diff
@@ -184,16 +194,45 @@ def _mk_field_attrs(field_names, extra_namespace):
                 break
 
 
-def _mk_field_property(field, attr):
-    def field_fn_getter(self):
-        return getattr(self, attr)
+@to_dict
+def _mk_field_props(field_names, field_attrs):
+    # we can't just define the getter in the loop as attr in the getter scope would not refer to
+    # the current value of the loop variable, but always the last one (see
+    # https://stackoverflow.com/q/233673)
+    def getter_factory(attr):
```

Comment:
coool 😎

Reply:
The standard library `operator.attrgetter(attr)` does exactly this.